sdk/src/pubkey.rs

```diff
@@ -79,6 +79,18 @@ impl Pubkey {
         ))
     }
 
+    pub fn create_program_address(
+        seeds: &[&str],
+        program_id: &Pubkey,
+    ) -> Result<Pubkey, PubkeyError> {
+        // Use a deterministic base for all program addresses that are owned by `program_id`
+        let mut base = Pubkey::create_with_seed(program_id, &"ProgramDerivedAddress", program_id)?;
```

Comment:
What does this function do?

Reply:
I think we should use hash directly, since it’s going to have impact on security.  Hash all the strings then haha the output.

Reply:
I think we should use hash directly, since it’s going to have impact on security.  Hash all the strings then hash the output.

Reply:
@aeyakovenko Is this more whast you had in mind?